src/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.RSA.cs

```diff
@@ -220,6 +252,94 @@ internal static partial class AppleCrypto
                 });
         }
 
+        private static bool ProcessPrimitiveResponse(
+            int returnValue,
+            SafeCFDataHandle cfData,
+            SafeCFErrorHandle cfError,
+            Span<byte> destination,
+            out int bytesWritten)
+        {
+            const int kErrorSeeError = -2;
+            const int kSuccess = 1;
+
+            if (returnValue == kErrorSeeError)
+            {
+                throw CreateExceptionForCFError(cfError);
+            }
+
+            if (returnValue == kSuccess && !cfData.IsInvalid)
+            {
+                return CoreFoundation.TryCFWriteData(cfData, destination, out bytesWritten);
+            }
+
+            Debug.Fail($"Unknown return value ({returnValue}) or no data object returned");
+            throw new CryptographicException();
+        }
+
+        internal static bool TryRsaDecryptionPrimitive(
+            SafeSecKeyRefHandle privateKey,
+            ReadOnlySpan<byte> source,
+            Span<byte> destination,
+            out int bytesWritten)
+        {
+            int returnValue = AppleCryptoNative_RsaDecryptionPrimitive(
+                privateKey,
+                ref MemoryMarshal.GetReference(source),
```

Comment:
Does the data need to be pinned while making the call?  Maybe that's already handled at a higher layer? (Same question for other such cases.)

Reply:
Oh, wait, you're passing a ref.  Nevermind.